### ISSUE TITLE
Unable to use docker container attached to macvlan network

### ISSUE THREAD

**Issue by i286** [codeproject/CodeProject.AI-Server#3]
Here is my compose file (using Portainer with existing macvlan network)

```
version: "3.9"

services:
  senseai:
    container_name: codeproject_ai
    image: codeproject/ai-server:latest
    environment:
      - TZ=Australia/Melbourne
    networks:
      dmz:
        ipv4_address: 192.168.67.112
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - codeproject-data:/usr/share/CodeProject/AI
    restart: unless-stopped

networks:
  dmz:
    external:
      name: vlan67_dmz

volumes:
  codeproject-data:
```

When I run this on my server it starts but checking the logs, it errors out, please note there is nothing else on the port of course:
```
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Overriding address(es) 'http://+:5000'. Binding to endpoints defined via IConfiguration and/or UseKestrel() instead.
fail: Microsoft.Extensions.Hosting.Internal.Host[9]
      BackgroundService failed
      System.Threading.Tasks.TaskCanceledException: A task was canceled.
         at CodeProject.AI.API.Server.Frontend.VersionProcessRunner.ExecuteAsync(CancellationToken stoppingToken)
         at Microsoft.Extensions.Hosting.Internal.Host.TryExecuteBackgroundServiceAsync(BackgroundService backgroundService)
crit: Microsoft.Extensions.Hosting.Internal.Host[10]
      The HostOptions.BackgroundServiceExceptionBehavior is configured to StopHost. A BackgroundService has thrown an unhandled exception, and the IHost instance is stopping. To avoid this behavior, configure this to Ignore; however the BackgroundService will not be restarted.
      System.Threading.Tasks.TaskCanceledException: A task was canceled.
         at CodeProject.AI.API.Server.Frontend.VersionProcessRunner.ExecuteAsync(CancellationToken stoppingToken)
         at Microsoft.Extensions.Hosting.Internal.Host.TryExecuteBackgroundServiceAsync(BackgroundService backgroundService)
info: Microsoft.Hosting.Lifetime[0]
      Application is shutting down...
fail: Microsoft.Extensions.Hosting.Internal.Host[9]
      BackgroundService failed
      System.Threading.Tasks.TaskCanceledException: A task was canceled.
         at CodeProject.AI.API.Server.Frontend.BackendProcessRunner.ExecuteAsync(CancellationToken stoppingToken)
         at Microsoft.Extensions.Hosting.Internal.Host.TryExecuteBackgroundServiceAsync(BackgroundService backgroundService)
crit: Microsoft.Extensions.Hosting.Internal.Host[10]
      The HostOptions.BackgroundServiceExceptionBehavior is configured to StopHost. A BackgroundService has thrown an unhandled exception, and the IHost instance is stopping. To avoid this behavior, configure this to Ignore; however the BackgroundService will not be restarted.
      System.Threading.Tasks.TaskCanceledException: A task was canceled.
         at CodeProject.AI.API.Server.Frontend.BackendProcessRunner.ExecuteAsync(CancellationToken stoppingToken)
         at Microsoft.Extensions.Hosting.Internal.Host.TryExecuteBackgroundServiceAsync(BackgroundService backgroundService)


Unable to start the server: Address family not supported by protocol.
Check that another instance is not running on the same port.
Press Enter to close.
```

**Comment by i286** [comment]
Just tried again with the latest image available as of right now and it is still same issue :(

The messages are slighty different but same error message as below
`Press Enter to close.      gInfor Server: ** Operating System: Linux 5.4.0-124-generic #140-Ubuntu SMP Thu Aug 4 02:23:37 UTC 2022
Infor Server: ** Architecture:     X64
Infor Server: ** Environment:
Infor Server: ** Platform:         docker
Infor Server: ** In Docker:        True
Infor Server: ** App DataDir:      /usr/share/CodeProject/AI
Infor Server: 
Debug Server: ------------------------------------------------------------------
Debug Server: Expanded ROOT_PATH       = /app
Debug Server: Expanded MODULES_PATH    = /app/AnalysisLayer
Debug Server: Expanded PYTHON_BASEPATH = /app/AnalysisLayer/bin/%PYTHON_RUNTIME%/venv
Debug Server: Expanded PYTHON_PATH     = %PYTHON_RUNTIME%
Debug Server: ------------------------------------------------------------------
Infor Server: BackendProcessRunner Start
Trace Server: Starting Background AI Modules
Error Server: BackgroundService failed
Error Server: BackgroundService failed


Unable to start the server: Address family not supported by protocol.
Check that another instance is not running on the same port.`

**Comment by ChrisMaunder** [comment]
Are you trying to access via http:// or https://? We only support http:// at the moment.

**Comment by vajonam** [comment]
The issue is that the internal server is looking to bind to the IPV6 which is disabled. There needs to be an option to bind only ipv4 addreses

**Comment by jdrozdnovak** [comment]
While solving this, could you point us to your dockerfiles, I am confident the community can solve this.

**Comment by i286** [comment]
I figured it out. I disable ipv6 as a matter of course. I updated another docker app i use and was checking why it wasnt starting up. it suddenly gave me errrors about dual stack not enabled and a very similar message from python as per my original issue.
Took me this long to put 2 and 2 together. I tested it by enabling ipv6 on this host and voila...

I can start the codeproject ai server stack with my original config!
